frontend/src/components/Navigator/index.js

```diff
@@ -143,6 +143,15 @@ function Navigator({
     const { status, ...values } = getValues();
     const data = { ...formData, ...values, pageState: newNavigatorState() };
     updateFormData(data);
+
+    // jest automatically sets NODE_ENV to test.
+    // Our tests basically assume the form is always dirty, so we bail out here
+    // only when we're not running in a test environment.
+    if (process.env.NODE_ENV !== 'test' && !isDirty) {
```

Comment:
I am still not a fan of checking for test env

Reply:
removed reliance on this in 7a0a9d8 with no change in behavior so, turns out, checking this wasn't even very useful